packages/steps/src/__tests__/BuildWorkflowValidator-test.ts

```diff
@@ -0,0 +1,233 @@
+import assert from 'assert';
+
+import { BuildStep } from '../BuildStep.js';
+import { BuildStepInput } from '../BuildStepInput.js';
+import { BuildStepOutput } from '../BuildStepOutput.js';
+import { BuildWorkflow } from '../BuildWorkflow.js';
+import { BuildWorkflowValidator } from '../BuildWorkflowValidator.js';
+import { BuildConfigError } from '../errors/BuildConfigError.js';
+import { BuildWorkflowError } from '../errors/BuildWorkflowError.js';
+
+import { createMockContext } from './utils/context.js';
+import { getError } from './utils/error.js';
+
+describe(BuildWorkflowValidator, () => {
+  test('non unique step ids', async () => {
+    const ctx = createMockContext();
+    const workflow = new BuildWorkflow({
+      buildSteps: [
+        new BuildStep(ctx, {
+          id: 'test1',
+          command: 'echo 123',
+          workingDirectory: ctx.workingDirectory,
+        }),
+        new BuildStep(ctx, {
+          id: 'test1',
+          command: 'echo 456',
+          workingDirectory: ctx.workingDirectory,
+        }),
+        new BuildStep(ctx, {
+          id: 'test1',
+          command: 'echo 789',
+          workingDirectory: ctx.workingDirectory,
+        }),
+        new BuildStep(ctx, {
+          id: 'test3',
+          command: 'echo 123',
+          workingDirectory: ctx.workingDirectory,
+        }),
+        new BuildStep(ctx, {
+          id: 'test3',
+          command: 'echo 456',
+          workingDirectory: ctx.workingDirectory,
+        }),
+      ],
+    });
+
+    const validator = new BuildWorkflowValidator(workflow);
+    const error = getError<BuildWorkflowError>(() => {
+      validator.validate();
+    });
+    expect(error).toBeInstanceOf(BuildWorkflowError);
+    assert(error instanceof BuildWorkflowError);
```

Comment:
```suggestion
    expect(error).toBeInstanceOf(BuildWorkflowError);
```

Wouldn't only one of these do the job?

Reply:
No
<img width="866" alt="Screenshot 2023-02-13 at 15 22 52" src="https://user-images.githubusercontent.com/5256730/218483718-34f1976e-9f65-40bb-b854-dea54d27a1c2.png">